conans/client/toolchain/cmake/__init__.py

```diff
@@ -0,0 +1,17 @@
+from conans.client.tools import cross_building
+from conans.errors import ConanException
+from .android import CMakeAndroidToolchain
+from .native import CMakeNativeToolchain
+
+
+def CMakeToolchain(conanfile, *args, **kwargs):
+    skip_x64_x86 = conanfile.settings.get_safe('os') in ['Windows', 'Linux']
+    if not cross_building(conanfile, skip_x64_x86=skip_x64_x86):
```

Comment:
I would simplify this, and add new logic only when the derived classes are really created. So at the moment I would have just 2, the Android one which is very clear, and another CMakeGenericToolchain() that could be now both cross-building and native, until someone actually does some cross compilation.